storage/local/interface.go

```diff
@@ -26,6 +26,8 @@ import (
 // are goroutine-safe. Storage implements storage.SampleAppender.
 type Storage interface {
 	prometheus.Collector
+	Querier
```

Comment:
Also embedding the Collector interface is not great. Doesn't have to be this issue but it has nothing to do with a storage.